reco_utils/evaluation/spark_diversity_evaluator.py

```diff
@@ -0,0 +1,362 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from pyspark.sql.types import *
+from pyspark.sql import functions as F
+
+
+class DiversityEvaluator:
```

Comment:
I think we need to maintain consistency between this file and the rest of the code. So several suggestions:
1. The other classes are called `SparkRatingEvaluation` and `SparkRankingEvaluation` [see code](https://github.com/microsoft/recommenders/blob/main/reco_utils/evaluation/spark_evaluation.py#L24), so I think we should rename this class to `DiversityEvaluation`.
2. The file `spark_evaluation.py` contains both `SparkRatingEvaluation` and `SparkRankingEvaluation`, whereas there is a new python file `spark_diversity_evaluator.py`, I think we should have either the 3 classes in the same file or have 3 files, one for each.

Reply:
item 1 is addressed.

Reply:
for item 2, I'd suggest to make it a separate PR